code/modules/ftl/machinery/phase_cannon.dm

```diff
@@ -0,0 +1,95 @@
+/obj/machinery/power/shipweapon //PHYSICAL WEAPON
+	name = "phase cannon"
+	desc = "A powerful weapon designed to take down shields.\n<span class='notice'>Alt-click to rotate it clockwise.</span>"
+	icon = 'icons/obj/96x96.dmi'
+	icon_state = "phase_cannon"
+	pixel_x = -32
+	pixel_y = -32
+	anchored = 0
+	density = 1
+
+	var/charge_rate = 200
+	var/current_charge = 0
+
+	use_power = ACTIVE_POWER_USE
+	idle_power_usage = 1000
+	active_power_usage = 20000
+
+	var/obj/item/weapon_chip/chip
+
+	/obj/item/weapon/circuitboard/machine/phase_cannon
+		name = "circuit board (Phase Cannon)"
+		build_path = /obj/machinery/power/shipweapon
+
+/obj/machinery/power/shipweapon/process()
+	if(stat & (BROKEN|MAINT))
+		return
+	if(!chip)
+		current_charge = 0
+	if(!active_power_usage || avail(active_power_usage)) //Is there enough power available
+		var/load = min((chip.charge_to_fire - current_charge), charge_rate)		// charge at set rate, limited to spare capacity
```

Comment:
Currently runtimes if there is no chip installed

Reply:
good catch